esp/templates/program/modules/availabilitymodule/availability_form.html

```diff
@@ -54,31 +147,130 @@
+                    </div>
+                  </th></tr>
+                  {% for selection in group.selections %}
+                  <tr><td nowrap="" class="proposed" onmousedown="md(this);" onmouseover="mo(this);" name="{{ selection.slot.id }}">
+                      <input type="checkbox" class="hidden" name="timeslots" value="{{ selection.slot.id }}" {% if selection.checked %}checked{% endif %} {% if selection.taken %}disabled{% endif %}>
+                      {{ selection.slot.start|date:"g:i A"}}
+                  </td></tr>
+                  {% endfor %}
+                </tbody>
+              </table>
+            </td>
+            {% if forloop.counter|divisibleby:2 %}</tr><tr><td></td></tr><tr>{% endif %}
+          {% endfor %}
+          </tr>
+        </tbody>
+      </table>
+
+    </div>
+    <div class="side">
+      <span class="summary" style="background-color: #EFEFEF;">Not Available</span>
+      <span class="summary" style="background-color: #00FF00;">Available</span>
+      <span class="summary" style="background-color: #42b3f4;">Scheduled to Teach</span>
+    </div>
+
+  </div>
+  <br>
+
+  <input type="submit" class="btn btn-primary" value="Submit" />
+</center>
+
+</form>
 
 </div>
 
+<script>
+var setting = false;
+var down = false;
+
+//Records mousedown
+function md(td) {
+  down = true;
+  setting = !isSet(td);
+  mo(td);
+}
+
+//Enables hightlighting multiple cells
+function mo(td) {
+  if (!down) return;
+  if (setting) {
+    on(td);
+  } else {
+    off(td);
+  }
+}
+
+function isSet(td) {
+  return td.className == "canDo" || td.className == "teaching";
+}
+
+//Activate checkbox
+function on(td) {
+  var checkbox = $j('input[value='+td.getAttribute("name")+']')[0]
+  if (td.className != "teaching") {
+  	td.className = "canDo";
+  	checkbox.checked = true;
+  }
+}
+
+//Deactivate checkbox
+function off(td) {
+  var checkbox = $j('input[value='+td.getAttribute("name")+']')[0]
+  if (!checkbox.disabled) {
+    td.className = "proposed";
+    checkbox.checked = false;
+  }
+}
+
+//Clicking the header turns the entire block on/off
+function header(e, col) {
+  var somethingToSet = false;
+  var block = document.getElementById("block_" + col);
+  for (var i = 1; i < block.rows.length; i++) {
+    var td = block.rows[i].cells[0];
+    if (!isSet(td)) somethingToSet = true;
+  }
+  for (var i = 1; i < block.rows.length; i++) {
+    var td = block.rows[i].cells[0];
+    if (somethingToSet) {
+      on(td);
+    } else {
+      off(td);
+    }
+  }
+}
+
+//Sets colors of cells based on status of checkboxes upon loading page
+$j(document).ready(function() {
+  $j('#grid input:checked').each(function(i, e) {
+    if (this.disabled == true) {
+      document.getElementsByName($j(this).attr('value'))[0].className = "teaching";
+      document.getElementsByName($j(this).attr('value'))[0].title = "You are currently scheduled to teach during this timeblock";
```

Comment:
"Timeblock" is jargon specific to our codebase and shouldn't be used in user-facing text. Replace it with "time".